base/expr.jl

```diff
@@ -2,12 +2,12 @@
 
 ## symbols ##
 
-symbol(s::Symbol) = s
-symbol(s::ASCIIString) = symbol(s.data)
-symbol(s::UTF8String) = symbol(s.data)
-symbol(a::Array{UInt8,1}) =
+Symbol(s::Symbol) = s
```

Comment:
As @nalimilan said, the default constructor calls `convert`, not the other way around, so if you remove something it should be the constructor(s).